subsys/bluetooth/mesh/sensor.c

```diff
@@ -61,21 +61,19 @@ bool bt_mesh_sensor_delta_threshold(const struct bt_mesh_sensor *sensor,
 		thrsh_mill = (prev_mill * thrsh_mill) / (100LL * 1000000LL);
 	}
 
-	if (BT_DBG_ENABLED) {
-		char delta_str[BT_MESH_SENSOR_CH_STR_LEN];
-		char curr_str[BT_MESH_SENSOR_CH_STR_LEN];
-		char prev_str[BT_MESH_SENSOR_CH_STR_LEN];
-		char thrsh_str[BT_MESH_SENSOR_CH_STR_LEN];
+	char delta_str[BT_MESH_SENSOR_CH_STR_LEN];
+	char curr_str[BT_MESH_SENSOR_CH_STR_LEN];
```

Comment:
Why is all debug functionality added unconditionally (here, in sensor_srv.c and lightness_srv.c)? What is the purpose\advantage to have this in real devices for customers apart from extra memory and MCU time consumption to handle this?